pyvsystems/contract.py

```diff
@@ -1,115 +1,290 @@
+                                                                                                              1) else Deser.parse_arrays(
+                state_map_bytes)
+            self.textual = Deser.parse_arrays(contract_bytes[state_map_end:len(contract_bytes)])
+        except ValueError or TypeError:
+            raise InvalidContractException("Contract is not initialized")
+
+
+def language_code_builder(code):
+    if len(code) == ContractMeta.language_code_byte_length:
+        language_code = Deser.serialize_string(code)
+        return language_code
+    else:
+        logging.error("Wrong language code length")
+        raise Exception("Wrong language code length")
+
+
+def language_version_builder(version):
+    if len(struct.pack(">I", version)) == ContractMeta.language_version_byte_length:
+        return struct.pack(">I", version)
+    else:
+        logging.error("Wrong language version length")
+        raise Exception("Wrong language code length")
+
+
+def bytes_builder_from_list(input_list):
+    if type(input_list) is list:
+        return Deser.serialize_array(Deser.serialize_arrays(input_list))
+    else:
+        logging.error("The input should be a list")
+
+def chain_system_contract_id(chain):
+    unhashedAddress = chr(6) + str(chain.chain_id) + hashChain(base58.b58encode(''))[0:20]
+    addressHash = hashChain(str2bytes(unhashedAddress))[0:4]
+    contract_id = bytes2str(base58.b58encode(str2bytes(unhashedAddress + addressHash)))
+    return contract_id
+
+def token_id_from_contract_id(contract_id, idx):
+    address_bytes = base58.b58decode(contract_id)
+    contract_id_no_check_sum = address_bytes[1:(len(address_bytes) - ContractMeta.check_sum_length)]
+    without_check_sum = struct.pack("b", ContractMeta.token_address_version) + contract_id_no_check_sum + struct.pack(
+        ">I",
+        idx)
+    return bytes2str(
+        base58.b58encode(without_check_sum + str2bytes(hashChain(without_check_sum)[0:ContractMeta.check_sum_length])))
+
+
+def serialize_data(data_entry_list):
+    custom_data_stack = []
+    if not type(data_entry_list) is list:
+        data_entry_list = [data_entry_list]
+    for data in data_entry_list:
+        custom_data_stack.append(data.bytes)
+    return Deser.serialize_array(custom_data_stack)
+
+
+def data_entry_from_base58_str(str_object):
+    base58_str = base58.b58decode(str_object)
+    return data_entries_from_bytes(base58_str)
+
+
+def data_entries_from_bytes(bytes_object):
+    length = struct.unpack(">H", bytes_object[0:2])[0]
+    all_data = []
+    pos_drift = 2
+    for pos in range(length):
+        [array_info, pos_drift] = parse_data_entry_array_size(bytes_object, pos_drift)
+        all_data.append(array_info)
+    return all_data
+
+
+def parse_data_entry_array_size(bytes_object, start_position):
+    if bytes_object[start_position: start_position + 1] == Type.public_key:
```

Comment:
`bytes_object[start_position: start_position + 1]` used multiple times, may pick it out (before if) and simplify the code